### PR TITLE
CURA-7868: Introduce the file provider plugin type

### PR DESCRIPTION
**Pull request by konskarm**
CURA-7868

### CODE REVIEW

**Review by Ghostkeeper — CHANGES_REQUESTED** [review]

File: resources/qml/Menus/FileMenu.qml [hunk 1]
```diff
@@ -22,6 +22,23 @@ Menu
...
+        visible: (CuraApplication.getFileProviderModel().count > 1)
+    }
+
+    Connections
+    {
+        target: CuraApplication.getFileProviderModel()
+        onItemsChanged:
+        {
+            openMenu.visible = (CuraApplication.getFileProviderModel().count == 1) // 1 because the open local files menu should always exist in the model
+            openFilesMenu.visible = (CuraApplication.getFileProviderModel().count > 1)
```
[the remark this answers is not in the document]
Comment (konskarm) [reply resources/qml/Menus/FileMenu.qml]: Your suspicion was correct! After using a variable for the model, the bindings were created automatically without the need for a Connection. Good catch!

Fixed in https://github.com/Ultimaker/Cura/pull/9012/commits/70550594cd113bee4af662c1dd980738ed3dbe0d.

File: resources/qml/TableView.qml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2021 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.10
```
Comment (jellespijker) [inline resources/qml/TableView.qml:4]: Latest supported QtQuick for Qt5.10 is 2.7

**Review by jellespijker — CHANGES_REQUESTED** [review]
Couple of minor changes

**Review by jellespijker — APPROVED** [review]

File: resources/qml/TableView.qml (hunk 2 above)
Comment (jellespijker) [reply resources/qml/TableView.qml:4]: Due to conflicting documentation on the Qt website, we will leave it as is and see how our build system handles this version

**Pull request merged**